test/micro/org/openjdk/bench/java/util/concurrent/UnparkBenchSleepersAfter.java

```diff
@@ -0,0 +1,142 @@
+ * questions.
+ */
+package org.openjdk.bench.java.util.concurrent;
+
+import org.openjdk.jmh.annotations.Benchmark;
+import org.openjdk.jmh.annotations.BenchmarkMode;
+import org.openjdk.jmh.annotations.Fork;
+import org.openjdk.jmh.annotations.Measurement;
+import org.openjdk.jmh.annotations.Mode;
+import org.openjdk.jmh.annotations.OutputTimeUnit;
+import org.openjdk.jmh.annotations.Param;
+import org.openjdk.jmh.annotations.Scope;
+import org.openjdk.jmh.annotations.Setup;
+import org.openjdk.jmh.annotations.State;
+import org.openjdk.jmh.annotations.TearDown;
+import org.openjdk.jmh.annotations.Threads;
+import org.openjdk.jmh.annotations.Warmup;
+
+import java.util.concurrent.BrokenBarrierException;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.CyclicBarrier;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.locks.LockSupport;
+
+@BenchmarkMode(Mode.Throughput)
+@OutputTimeUnit(TimeUnit.SECONDS)
+@State(Scope.Benchmark)
+@Fork(1)
+@Threads(1)
+@Warmup(iterations =  5, time = 5)
+@Measurement(iterations =  5, time = 5)
+public class UnparkBenchSleepersAfter {
+
+    /*
+        This micro creates thousands of sleeper threads after the threads doing the barrier await
+        to see if that has any effect on the barrier performance.
+     */
+
+    @Param({"4000"})
+    int idles;
+
+    @Param({"2"})
+    int workers;
+
+    CyclicBarrier barrier;
+
+    @Benchmark
+    public void barrier() throws InterruptedException {
+        CountDownLatch latch = new CountDownLatch(workers);
+        for (int i = 0; i < workers; i++) {
+            exec.submit(() ->
+            {
+                try {
+                    barrier.await();
+                } catch (InterruptedException | BrokenBarrierException e) {
+                    barrier.reset();
+                } finally {
+                    latch.countDown();
+                }
+            });
+        }
+        latch.await();
+    }
+
+    IdleThread[] idle_threads;
+
+
+    ExecutorService exec;
+
+    @Setup
+    public void setup() throws InterruptedException {
+        barrier = new CyclicBarrier(workers);
+        exec = Executors.newFixedThreadPool(workers);
+        CountDownLatch latch = new CountDownLatch(workers);
+        for (int i = 0; i < workers; i++) { // warmup exec
+            exec.submit(() ->
+            {
+                try {
+                    Thread.sleep(0);
+                } catch (InterruptedException e) {
+                    throw new RuntimeException(e);
+                } finally {
+                    latch.countDown();
+                }
+            });
+        }
+        latch.await();
+        idle_threads = new IdleThread[idles];
+        for(int i=0; i < idle_threads.length; i++) {
+            new Thread(idle_threads[i] = new IdleThread()).start();
+        }
```

Comment:
These `idleThreads` are not actually `Thread`-s, they are `Runnable`-s.

```suggestion
        for(int i = 0; i < idles; i++) {
            Runnable r = new IdleRunnable();
            idleRunnables[i] = r;
            new Thread(r).start();
        }
```